### ISSUE TITLE
GPG sign on Windows?

### ISSUE THREAD

**Issue by hadrienbecle** [ForkIssues/TrackerWin#1833]
Hey,

Sorry about the question, but I couldn't find how to sign commits with GPG on Windows.

I understand there should be a GPG tab in the settings repository, at least on macOS (screenshots in fork-dev/Tracker#20). Does this feature exist on Windows?

Thanks,

**Comment by DanPristupov** [comment]
No, there's no similar UI on Windows. Currently, you must configure GPG in the command line.

**Comment by hadrienbecle** [comment]
Okay, that's what I thought :/ I configured GPG and told Fork which git instance to use, and it works perfectly. Thanks for the (really fast) answer!

Is the feature on the roadmap?

**Comment by carun** [comment]
We are discussing about making the GPG signature mandatory in our company (of course most of us use Fork client). So please consider implementing this feature.

**Comment by carun** [comment]
FWIW, we use https://gpg4win.org/index.html on Windows.

**Comment by Xeevis** [comment]
> We are discussing about making the GPG signature mandatory in our company (of course most of us use Fork client). So please consider implementing this feature.
> FWIW, we use https://gpg4win.org/index.html on Windows.

What stops you? You can sign with Fork easily.
1. Generate key-pair in Kleopatra (from Gpg4win)
2. Add public key under your github account
3. Create file `%homepath%/.gitconfig`
```
[user]
    email = YourName@users.noreply.github.com
    name = YourName
    signingkey = <Key-ID>
[core]
    autocrlf = true
    sshCommand = 'C:\\Windows\\System32\\OpenSSH\\ssh.exe'
[gpg]
    program = C:\\Program Files (x86)\\GnuPG\\bin\\gpg.exe
[commit]
    gpgsign = true
```

**Comment by hossain-khan** [comment]
> > We are discussing about making the GPG signature mandatory in our company (of course most of us use Fork client). So please consider implementing this feature.
> > FWIW, we use https://gpg4win.org/index.html on Windows.
> 
> What stops you? You can sign with Fork easily.
> 
> 1. Generate key-pair in Kleopatra (from Gpg4win)
> 2. Add public key under your github account
> 3. Create file `%homepath%/.gitconfig`
> 
> ```
> [user]
>     email = YourName@users.noreply.github.com
>     name = YourName
>     signingkey = <Key-ID>
> [core]
>     autocrlf = true
>     sshCommand = 'C:\\Windows\\System32\\OpenSSH\\ssh.exe'
> [gpg]
>     program = C:\\Program Files (x86)\\GnuPG\\bin\\gpg.exe
> [commit]
>     gpgsign = true
> ```

I can confirm that this works seamlessly using https://www.gpg4win.org/ for windows and configuration mentioned above.

Here is what I have done.
1. Install the gpg4win
2. Open Kleopatra app and generate new key
3. Add the public key to github account (see this [guide](https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key))
4. Use the config mentioned above
5. Make a new commit - which asks for the GPG key to sign using Kelopatra (app runs in background)
6. ✅ 

![Screenshot 2024-03-25 090931](https://github.com/fork-dev/TrackerWin/assets/99822/d3781455-e6a1-429a-826a-1491c13a45e9)

**Comment by d3xMachina** [comment]
Don't set the gpg program path in .gitconfig if you use Fork. It comes preinstalled. So it won't find the keys if you set the path to another gpg installation.

**Comment by ChrisKulesza** [comment]
> Okay, that's what I thought :/ I configured GPG and told Fork which git instance to use, and it works perfectly. Thanks for the (really fast) answer!
> 
> Is the feature on the roadmap?

~~Is that all you need to do for Fork to realise that there is a GPG key? I have now created one and stored the public key in GitHub. Does Fork handle the rest independently? I can't find a specific setting for signing the commits. I am using Windows.~~

I asked my question prematurely. I think I have understood it now. I entered the entry in the .gitconfig with the following command from the GitHub Docs. It remains to be seen whether I have to explicitly specify the paths to ssh.exe and gpg.exe.

```shell
git config --global user.signingkey <GPG Key ID>
```

**Comment by DefinitelyADev** [comment]
Would be nice to have the same functionality on windows.
